### PR TITLE
Skip compiler unload/reload tests if compiler is not available

### PR DESCRIPTION
**Pull request by alycm**
Note that tests that use the compiler helper functions but won't work without a
compiler are in the `subtests_to_skip_with_offline_compiler` list.  I didn't do
that here because they tests directly use `clBuildProgram` etc directly,
because they're specifically testing it's behaviour in edge-cases.

### CODE REVIEW

File: test_conformance/compiler/test_feature_macro.cpp [hunk 1]
```diff
@@ -724,21 +724,15 @@ int test_consistency_c_features_list(cl_device_id deviceID,
...
-    }
 
+    // Note: Not checking that the feature array is empty for the compiler not
+    // available case because the specification says "For devices that do not
+    // support compilation from OpenCL C source, this query may return an empty
+    // array."  It "may" return an empty array implies that an implementation
+    // also "may not".
+    check_compiler_available(deviceID);
+
+    cl_int error = CL_SUCCESS;
```
Comment (bashbaug) [inline test_conformance/compiler/test_feature_macro.cpp:735]: Minor: since this value represents a test status and not an OpenCL error code now, it more accurately should be an `int` and should be initialized to `TEST_PASS`.  This will not cause any issues in practice as-is.

**Review by bashbaug — APPROVED** [review]
LGTM - just one minor comment that can be fixed later (or never).

File: test_conformance/compiler/test_feature_macro.cpp (hunk 1 above)
Comment (alycm) [reply test_conformance/compiler/test_feature_macro.cpp]: Makes sense, fixed.

**Review by bashbaug — APPROVED** [review]
LGTM - thanks!

**Review by gwawiork — APPROVED** [review]
LGTM

**Review by kpet — APPROVED** [review]
